### PR TITLE
Fixed config.json download to go to user-supplied cache directory

### PR DESCRIPTION
**Pull request by ulatekh**
Before, config.json would end up in the default cache directory. Now everything but version.txt is placed properly, and since it's downloaded when pipeline is imported, fixing that will be more complex.

I ran "pytest --picked"; it generated 3 warnings, none of which seemed to have anything to do with my change.

@Narsil since this is related to pipeline.

### CODE REVIEW

File: src/transformers/pipelines/__init__.py [hunk 1]
```diff
@@ -785,6 +789,10 @@ def pipeline(
                 **hub_kwargs,
```
Comment (amyeroberts) [inline src/transformers/pipelines/__init__.py:789]: Why not just:

```suggestion
                cache_dir=model_kwargs.get("cache_dir"),
                **hub_kwargs,
```

**Review by amyeroberts — COMMENTED** [review]
Thanks for opening an PR and fixing this issue! 

Just a suggestion which removes the need to add then remove values into hub_kwargs

File: src/transformers/pipelines/__init__.py (hunk 1 above)
Comment (ulatekh) [reply src/transformers/pipelines/__init__.py:789]: Would that work properly if there is no cache_dir in model_kwargs? Keep in mind that I've learned and forgotten Python twice, and I guess I'm on round 3.

Comment (amyeroberts) [reply src/transformers/pipelines/__init__.py:789]: Yep! `get` is the safe way to access a value from a dictionary e.g.:

```
d = {'a': 1, 'b': 2, 'c': 3}

# This fails
d['d']

# This returns None
d.get('d')
```

**Review by amyeroberts — APPROVED** [review]
Thanks for iterating and adding this fix!

**Pull request merged**